WorkloadManagementSystem/Utilities/PilotWrapper.py

```diff
@@ -236,7 +248,7 @@ def getPilotFilesCompressedEncodedDict(pilotFiles, proxy=None):
 
   for pf in pilotFiles:
     with open(pf, "r") as fd:
-      pfContent = fd.read()
+      pfContent = fd.read().decode()
```

Comment:
I think you have to not use decode() here, and add encode() in the compress, like the changes in the Test_pilotWrapper?

Reply:
You mean that I should do `open(pf, "rb") as fd` ?

Reply:
This is all only for python3, python2 seems to be much more relaxed about this.

I meant:
```python
    with open(pf, "r") as fd:
      pfContent = fd.read()
    pfContentEncoded = base64.b64encode(bz2.compress(pfContent.encode(), 9))
```
(like in tests/Integration/WorkloadManagementSystem/Test_GenerateAndExecutePilotWrapper.py line 78 : https://github.com/DIRACGrid/DIRAC/pull/4392/files#diff-a208ebc972b1ee85374d34f8bd0389f4R78 ) (I referred to the wrong test file before.)

With 'rb' your have to do

```python
    with open(pf, "rb") as fd:
      pfContent = fd.read()
    pfContentEncoded = base64.b64encode(bz2.compress(pfContent, 9))
```
Or
```python
    with open(pf, "rb") as fd:
      pfContent = fd.read().decode()
    pfContentEncoded = base64.b64encode(bz2.compress(pfContent.encode(), 9))
```

Here is a gist: https://gist.github.com/andresailer/3538baf9cc84848c8e92e54ad11df691 that does all combinations of r/rb, decode, encode...
```
lxplus700:~$ python3 testDecode.py

Decoding: Error 'rb' TypeError("a bytes-like object is required, not 'str'",)

Decoding: Error 'r' AttributeError("'str' object has no attribute 'decode'",)

Nothing:  'rb' b'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI='

Nothing: Error 'r' TypeError("a bytes-like object is required, not 'str'",)

Encoding: Error 'rb' AttributeError("'bytes' object has no attribute 'encode'",)

Encoding:  'r' b'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI='

Both: 'rb' b'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI='

Both: Error 'r' AttributeError("'str' object has no attribute 'decode'",)
```

python2
```
lxplus700:~$ python2 testDecode.py
()
("Decoding:  'rb'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Decoding:  'r'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Nothing:  'rb'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Nothing:  'r'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Encoding:  'rb'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Encoding:  'r'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Both: 'rb'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
()
("Both: 'r'", 'QlpoOTFBWSZTWZlRxlkAAAJRgAAQQAA0DAgAIAAxDAgSjRkXtKOK4uCGcYhtHi7kinChITKjjLI=')
```